### ISSUE TITLE
Next.js server start hangs

### ISSUE THREAD

**Issue by eyaleizenberg** [bahmutov/start-server-and-test#348]
I created a new next.js project and added start-server-and-test but no matter what I do, the test run doesn't start. This is my output:

```
start-server-and-test http-get://localhost:3000 cypress.run

1: starting server using command "npm run start"
and when url "[ 'http-get://localhost:3000' ]" is responding with HTTP status code NaN
running tests using command "npm run cypress.run"


> rhino-next@0.1.0 start
> next start

ready - started server on 0.0.0.0:3000, url: http://localhost:3000
```

I ran the curl command to check if a Head request returns 200 and it does.

```
curl --head http://localhost:3000/     
HTTP/1.1 200 OK
X-Powered-By: Next.js
ETag: "dz0spzxup038i"
Content-Type: text/html; charset=utf-8
Content-Length: 4199
Vary: Accept-Encoding
Date: Tue, 03 Jan 2023 08:12:49 GMT
Connection: keep-alive
Keep-Alive: timeout=5
```

My guess is the part the says `is responding with HTTP status code NaN` is the problem.

What am I missing here?

Thanks.

**Comment by tvsbrent** [comment]
I am seeing this problem as well. I'm using `http-server` for hosting my test site in this case.

**Comment by dariobarrio** [comment]
In case can be of use for someone, I had the exactly same problem (in local) that what @eyaleizenberg was reporting.

I downgrade from Node v18.13.0 to v16.19.0 and it works well again. Something there is not quite right.

**Comment by dkonasov** [comment]
Faced this issue as well. I was launching my tests with command

`start-server-and-test 'npx http-server ./storybook-build -s -c-1 -p 8080' http://localhost:8080 'jest --config ./jest.e2e.config.js'`

and my test haven't started. For me issue was resolved when I changed localhost to 127.0.0.1:

`start-server-and-test 'npx http-server ./storybook-build -s -c-1 -p 8080' http://127.0.0.1:8080 'jest --config ./jest.e2e.config.js'`

However, when I tried to connect to my storybook app on http://localhost:8080 during the first run, everything was fine, site was available by domain name.

My node version is 18.12.0, start-server-and-test version is 1.15.2

**Comment by mikansc** [comment]
A workaround that worked for me is to using the local IP address instead localhost on the starting script:

`"ci": "start-server-and-test build-and-serve http://127.0.0.1:5000 cy:open"`

See:
[https://github.com/bahmutov/start-server-and-test/issues/333#issuecomment-1246163291](https://github.com/bahmutov/start-server-and-test/issues/333#issuecomment-1246163291)
[https://github.com/jeffbski/wait-on/issues/109#issue-1071849300](https://github.com/jeffbski/wait-on/issues/109#issue-1071849300)

**Comment by MikeMcC399** [comment]
- I submitted https://github.com/bahmutov/start-server-and-test/issues/349 as a separate issue

**Comment by mihailgaberov** [comment]
I had the same problem in a Meteor.js app. I fixed it by adding `--expect 200` and passing explicitly the port that is used by Cypress UI to the server. The `--expect` param replaces **NaN** with **200**, but without passing the port number the Cypress UI doesn't start at all.

Here is part of my package.json:
```
"start:e2e": "set METEOR_LOCAL_DIR=.meteor/test && meteor --port 4000",
"test:e2e": "start-test --expect 200 start:e2e :4000 cypress:ui"
```

**Comment by bahmutov** [comment]
Can someone please provide a small repo with a reproducible example?

**Comment by MikeMcC399** [comment]
@mihailgaberov 

> The --expect param replaces NaN with 200,

This issue is solved in [v1.15.4](https://github.com/bahmutov/start-server-and-test/releases/tag/v1.15.4).

**Comment by MikeMcC399** [comment]
- I expect this is the same root cause as https://github.com/cypress-io/github-action/issues/802 if the issue generally happens on GitHub running the default version of Node.js (currently 18.14.1). Judging by the other issue, running locally on any Node.js version there is no problem. On GitHub running Node.js 16 and earlier there is also no problem.

- `Next.js` uses `react` and the react server only listens on the IPv4 address `127.0.0.1` however due to changes in Node.js&nbsp;>=&nbsp;17 the dns lookup returns the IPv6 address `::1` on GitHub, so the wait-on test is trying on  `::1` and react is waiting on `127.0.0.1`. The two never meet up.

**Comment by MikeMcC399** [comment]
I suggest to open a separate issues for each other server that has been mentioned here, that is ones which are not Next.js, otherwise troubleshooting in this issue is going to get quite confusing!

@eyaleizenberg Are you still having an issue with Next.js?

**Comment by bahmutov** [comment]
Can someone provide a repo with an example that shows the problem? Or debug logs? Anything?

**Comment by MikeMcC399** [comment]
@bahmutov 

> Can someone provide a repo with an example that shows the problem? Or debug logs? Anything?

- See issue #356 which provides everything you asked for using an out-of-the-box react app (`npm create-react react-app-test`) with `start-server-and-test` and Cypress added.
